mappings/platform_file_centric.json

```diff
@@ -0,0 +1,314 @@
+          "tokenizer": "keyword"
+        }
+      },
+      "filter": {
+        "edge_ngram": {
+          "max_gram": 20,
+          "min_gram": 1,
+          "side": "front",
+          "type": "edge_ngram"
+        }
+      }
+    }
+  },
+  "mappings": {
+    "dynamic": false,
+    "date_detection": false,
+    "properties": {
+      "file_autocomplete": {
+        "fields": {
+          "analyzed": {
+            "analyzer": "autocomplete_analyzer",
+            "search_analyzer": "lowercase_keyword",
+            "type": "text"
+          },
+          "lowercase": {
+            "analyzer": "lowercase_keyword",
+            "type": "text"
+          },
+          "prefix": {
+            "analyzer": "autocomplete_prefix",
+            "search_analyzer": "lowercase_keyword",
+            "type": "text"
+          }
+        }
+      },
+      "program_id": {
+        "type": "keyword",
+        "copy_to": ["file_autocomplete"]
+      },
+      "primary_site": {
+        "type": "keyword"
+      },
+      "object_id": {
+        "type": "keyword",
+        "copy_to": ["file_autocomplete"]
+      },
+      "file_type": {
+        "type": "keyword"
+      },
+      "data_type": {
+        "type": "keyword",
+        "copy_to": ["file_autocomplete"]
+      },
+      "file_access": {
+        "type": "keyword"
+      },
+      "analysis": {
+        "properties": {
+          "analysis_id": {
+            "type": "keyword",
+            "copy_to": ["file_autocomplete"]
+          },
+          "analysis_type": {
+            "type": "keyword"
+          },
+          "analysis_version": {
+            "type": "integer"
+          },
+          "workflow": {
+            "properties": {
+              "workflow_name": {
+                "type": "keyword"
+              },
+              "workflow_version": {
+                "type": "keyword"
+              }
+            }
+          },
+          "experiment": {
+            "type": "object",
+            "properties": {
+              "platform": {
+                "type": "keyword"
+              },
+              "library_strategy": {
+                "type": "keyword"
+              }
+            }
+          }
+        }
+      },
+      "file": {
+        "properties": {
+          "id": {
+            "type": "keyword"
+          },
+          "size": {
+            "type": "integer"
+          },
+          "md5sum": {
+            "type": "keyword"
+          },
+          "name": {
+            "type": "keyword"
+          }
+        }
+      },
+      "index_file": {
+        "properties": {
+          "object_id": {
+            "type": "keyword"
+          },
+          "format": {
+            "type": "keyword"
+          },
+          "md5sum": {
+            "type": "keyword"
+          },
+          "name": {
+            "type": "keyword"
+          },
+          "size": {
+            "type": "integer"
+          }
+        }
+      },
+      "donors": {
```

Comment:
for the clinical data here since now we will have multiple primary diagnoses, some of the followups/specimens/treatments won't be relevant right?

Reply:
im not sure what you mean by `wont be relevant`?

Reply:
so my understanding was that we will index the primary diagnosis and all follow ups and treatments when we merge the file and the donor
but now we have multiple primary diagnoses in few cases where not all follow ups and specimens are related

Reply:
well, correct if i'm wrong but there are 2 options here once the dictionary has been changed: 
1. leave the `pd`, `follow-up`, and `specimen` where they are and add in the denormalized id links as part of the object (primary_diagnosis_id on treatment, follow_up, and specimen) 
2. Try and make a nested structure from primary diagnosis, holding objects below that that are related.  In this scenario, if there is then a follow-up that is not related to a primary_diagnosis or followup, then we would have to have a second section for that.  

Personally, I think that 1) makes more sense as 2) would be confusing to see two areas in the index named the same?

Reply:
Discussed with @blabadi - will go with 1. When we add the multi PD intot he dictionary we need to add the PD id in specimen, follow up treatment as well here in the schema.